src/main/java/no/nav/finnkandidatapi/kafka/harTilretteleggingsbehov/HarTilretteleggingsbehovProducer.java

```diff
@@ -93,21 +111,74 @@ public void kandidatSlettet(KandidatSlettet event) {
     public void permitteringEndretEllerOpprettet(PermittertArbeidssokerEndretEllerOpprettet event) {
         PermittertArbeidssoker permittertArbeidssoker = event.getPermittertArbeidssoker();
         Optional<Kandidat> kandidat = kandidatService.hentNyesteKandidat(permittertArbeidssoker.getAktørId());
+        Optional<Vedtak> vedtak = vedtakService.hentNyesteVedtakForAktør(permittertArbeidssoker.getAktørId());
         List<String> kategorier = kandidat.map(Kandidat::kategorier).orElse(Collections.emptyList());
 
-        lagOgSendMelding(permittertArbeidssoker.getAktørId(), kategorier, Optional.of(permittertArbeidssoker));
+        lagOgSendMelding(permittertArbeidssoker.getAktørId(), kategorier, Optional.of(permittertArbeidssoker), vedtak);
     }
 
-    private void lagOgSendMelding(String aktørId, List<String> kategorier, Optional<PermittertArbeidssoker> permittertArbeidssoker) {
+    private void lagOgSendMelding(String aktørId,
+                                  List<String> kategorier,
+                                  Optional<PermittertArbeidssoker> permittertArbeidssoker,
+                                  Optional<Vedtak> vedtak) {
         boolean harBehov = !kategorier.isEmpty();
-        List<String> kategorierOgPermittering = kombiner(kategorier, permittertArbeidssoker);
+        boolean erPermittert = sjekkOmErPermittert(permittertArbeidssoker, vedtak);
+        List<String> kategorierOgPermittering = kombiner(kategorier, erPermittert);
         HarTilretteleggingsbehov melding = new HarTilretteleggingsbehov(aktørId, harBehov, kategorierOgPermittering);
         sendKafkamelding(melding);
     }
 
-    private List<String> kombiner(List<String> kategorier, Optional<PermittertArbeidssoker> permittertArbeidssoker) {
+    private boolean sjekkOmErPermittert(Optional<PermittertArbeidssoker> permittertArbeidssoker, Optional<Vedtak> vedtak) {
```

Comment:
Mulig du hadde tenkt å gjøre dette allerede, men kunne vel flyttet all logikken for å velge riktig permittertstatus til en egen klasse